source/codegen/service_helpers.py

```diff
@@ -110,7 +127,26 @@ def get_array_param_size(parameter) -> str:
     return ''
 
 
-def create_param(parameter):
+def expand_varargs_parameters(parameters):
+    if not common_helpers.has_varargs_parameter(parameters):
+        return parameters
+    # omit the varargs parameter that we're going to expand
+    new_parameters = parameters[:-1]
+    varargs_parameter = parameters[-1]
+    assert common_helpers.is_varargs_parameter(varargs_parameter)
+    max_length = varargs_parameter['max_length']
+    # Many (all?) functions that take varargs take the first set of parameter as
```

Comment:
I think the way you're talking about `var_args` is a little too specific. But it might not matter. The other usage of `var_args` in the DAQ API are for the generic get/set attribute methods, which don't have this convention. But we're going to handle that by expanding the metadata to fill in specific types for the args.

I'd been calling these "more channels and states" methods, which is not that good a name either.

I think it's fine to mostly keep your code as-is. And comments like this can clarify that we're implementing a specific var_args convention where it's an arbitrary repetition of fixed args. And it would probably also work  an arbitrary repetition of args that aren't in the fixed param list (with changes; but we haven't implemented it).

I think it's also reasonable to think of what you're doing as a more specific convention `repeating_var_args` and potentially use that as structure for how it's represented in the metadata. Like the fixed-args could have a tag because they participate in that specific convention. If that's helpful.

Reply:
Yeah, I like this. I moved to the "repeated_var_args" convention and added a "repeating_var_arg" tag for the first copy of each argument.